save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/FileUploader.kt

```diff
@@ -32,47 +31,50 @@ import react.dom.html.ReactHTML.span
 import react.dom.html.ReactHTML.strong
 import react.dom.html.ReactHTML.ul
 
+import kotlinx.browser.window
 import kotlinx.js.jso
 
-/**
- * Component used to upload file
- */
 val fileUploader = fileUploader()
 
 /**
  * A component for file icon that changes depending on executable flag
  */
 @Suppress("TYPE_ALIAS", "EMPTY_BLOCK_STRUCTURE_ERROR")
 internal val fileIconWithMode: FC<FileIconProps> = FC { props ->
-    span {
-        className = ClassName("fa-layers mr-3")
-        title = "Click to mark file ${if (props.fileInfo.isExecutable) "regular" else "executable"}"
-        asDynamic()["data-toggle"] = "tooltip"
-        asDynamic()["data-placement"] = "top"
-        // if file was not executable, after click it will be; and vice versa
-        onClick = { _ ->
-            // hide previous tooltip, otherwise it gets stuck during re-render
-            val jquery = kotlinext.js.require("jquery")
-            jquery("[data-toggle=\"tooltip\"]").tooltip("hide")
-            props.onExecutableChange(props.fileInfo, !props.fileInfo.isExecutable)
-        }
-        onDoubleClick = {}
-        val checked = props.fileInfo.isExecutable
-        fontAwesomeIcon(icon = faFile, classes = "fa-2x") {
-            if (checked) {
-                asDynamic()["color"] = "Green"
-            }
-        }
+    if (props.fileInfo is FileInfo) {
```

Comment:
But in that case we will need one more check - that we really do pass `FileInfo` in order to make sure we didn't pass the wrong flag by default.
In this way it might be better to store it as state🤔

Reply:
Moreover, using `is` everywhere allows smart casts

Reply:
Decided to add such a flag huh. Done